dask_sql/physical/rel/logical/aggregate.py

```diff
@@ -48,18 +55,42 @@ class AggregationSpecification:
     """
     Most of the aggregations in SQL are already
     implemented 1:1 in dask and can just be called via their name
-    (e.g. AVG is the mean). However sometimes those already
-    implemented functions only work well for numerical
-    functions. This small container class therefore
+    (e.g. AVG is the mean). However sometimes those
+    implemented functions only work well for some native data types.
+    This small container class therefore
     can have an additional aggregation function, which is
-    valid for non-numerical types.
+    valid for non-supported native types.
     """
 
-    def __init__(self, numerical_aggregation, non_numerical_aggregation=None):
-        self.numerical_aggregation = numerical_aggregation
-        self.non_numerical_aggregation = (
-            non_numerical_aggregation or numerical_aggregation
-        )
+    def __init__(self, native_aggregation, custom_aggregation=None):
+        self.native_aggregation = native_aggregation
+        self.custom_aggregation = custom_aggregation or native_aggregation
+
+    def get_supported_aggregation(self, series):
+        native_agg = self.native_aggregation
+
+        # native_aggreagations work well for all numeric types
+        if pd.api.types.is_numeric_dtype(series.dtype):
+            return native_agg
+
+        # Todo: Add Categorical when support comes to dask-sql
+        if native_agg in ["min", "max"]:
+            if pd.api.types.is_datetime64_any_dtype(series.dtype):
+                return native_agg
+
+            if pd.api.types.is_string_dtype(series.dtype):
+                ## If dask_cudf strings dtype return native aggregation
+                if dask_cudf is not None and isinstance(series, dask_cudf.Series):
+                    return native_agg
+
+                ## If pandas StringDtype native aggregation works
+                ## and with ObjectDtype and Nulls native aggregation can fail
+                if isinstance(series, dd.Series) and isinstance(
```

Comment:
Essentially we need to take the  native agg code path for `Min/Max` if the `series` is either `dask_cudf` with `StringDtype` or `dask.DataFrame` with `pd.StringDtype` .   We want to take the custom agg code path if the dtype is an `object` .  



This is because `.min/max` on an object dtype series with nulls in pandas fails while it works for `StringDtype`   in pandas and works well for `cuDF` too. 
 

```python
#### Works
pd.Series(['a',None],dtype='string').min()
```
#### Fails
```
pd.Series(['a',None],dtype='object').min()

```

I wonder if we can have a cleaner combined check in `cuDF` . CC: @galipremsagar , @ayushdg  for thoughts ?